File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
    for i := 0; i < len(nums) - 1; i++ {
        needed := target - nums[i]
        for j := i + 1; j < len(nums); j++ {
            if nums[j] == needed {
                return []int{i, j}
            }
        }
    }
    return nil
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.java
class Solution {
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> curr = new ArrayList<>();
        dfs(nums, res, curr, 0, 0, target);
        return res; 
    }

    private void dfs(int[] nums, List<List<Integer>>  res, List<Integer> curr, int i, int sum, int target) {
        if (sum == target) {
            res.add(new ArrayList<>(curr));
            return;
        }

        if (i >= nums.length || sum > target) return;

        curr.add(nums[i]);
        dfs(nums, res, curr, i, sum + nums[i], target);
        curr.remove(curr.size() - 1);
        dfs(nums, res, curr, i + 1, sum, target);
    }
}
